opensearch/src/main/java/org/opensearch/sql/opensearch/data/value/OpenSearchExprValueFactory.java

```diff
@@ -83,39 +87,56 @@ public class OpenSearchExprValueFactory {
 
   private final Map<ExprType, Function<Content, ExprValue>> typeActionMap =
       new ImmutableMap.Builder<ExprType, Function<Content, ExprValue>>()
-          .put(INTEGER, c -> new ExprIntegerValue(c.intValue()))
-          .put(LONG, c -> new ExprLongValue(c.longValue()))
-          .put(SHORT, c -> new ExprShortValue(c.shortValue()))
-          .put(BYTE, c -> new ExprByteValue(c.byteValue()))
-          .put(FLOAT, c -> new ExprFloatValue(c.floatValue()))
-          .put(DOUBLE, c -> new ExprDoubleValue(c.doubleValue()))
-          .put(STRING, c -> new ExprStringValue(c.stringValue()))
-          .put(BOOLEAN, c -> ExprBooleanValue.of(c.booleanValue()))
-          .put(TIMESTAMP, this::parseTimestamp)
-          .put(DATE, c -> new ExprDateValue(parseTimestamp(c).dateValue().toString()))
-          .put(TIME, c -> new ExprTimeValue(parseTimestamp(c).timeValue().toString()))
-          .put(DATETIME, c -> new ExprDatetimeValue(parseTimestamp(c).datetimeValue()))
-          .put(OPENSEARCH_TEXT, c -> new OpenSearchExprTextValue(c.stringValue()))
-          .put(OPENSEARCH_TEXT_KEYWORD, c -> new OpenSearchExprTextKeywordValue(c.stringValue()))
-          .put(OPENSEARCH_IP, c -> new OpenSearchExprIpValue(c.stringValue()))
-          .put(OPENSEARCH_GEO_POINT, c -> new OpenSearchExprGeoPointValue(c.geoValue().getLeft(),
-              c.geoValue().getRight()))
-          .put(OPENSEARCH_BINARY, c -> new OpenSearchExprBinaryValue(c.stringValue()))
+          .put(OpenSearchDataType.of(OpenSearchDataType.Type.Integer),
```

Comment:
how about define the parse() funcion for each OpenSearchDataType?

Reply:
I wanted to avoid creating N inheritors of `OpenSearchDataType` - one per each `OpenSearch` type